contrib/stockfish/template.py

```diff
@@ -0,0 +1,57 @@
+pkgname = "stockfish"
+pkgver = "16.1"
+pkgrel = 0
+build_wrksrc = "src"
+build_style = "makefile"
+make_cmd = "gmake"
+make_build_target = "build"
+make_build_args = ["optimize=no"]
+hostmakedepends = ["gmake"]
+pkgdesc = "Free UCI chess engine derived from Glaurung"
+maintainer = "shtayerc <david.murko@mailbox.org>"
+license = "GPL-3.0-or-later"
+url = "https://github.com/official-stockfish"
+_net_file_big = "nn-b1a57edbea57.nnue"
+_net_file_small = "nn-baff1ede1f90.nnue"
+source = [
+    f"{url}/Stockfish/archive/sf_{pkgver}.tar.gz",
+    f"!https://tests.stockfishchess.org/api/nn/{_net_file_big}",
+    f"!https://tests.stockfishchess.org/api/nn/{_net_file_small}",
+]
+sha256 = [
+    "a5f94793b5d4155310397ba89e9c4266570ef0f24cd47de41a9103556f811b82",
+    "b1a57edbea574ca8b88d6837473845791beb53d885f87f86d5ccdd5659fbf3b2",
+    "baff1ede1f90c1dd1b4f772f1eff29848821801e8186345da7f0eb4121bd6f63",
+]
+
+tool_flags = {"LDFLAGS": ["-lpthread", "-Wl,-z,stack-size=2097152"]}
+# no check target
+options = ["!check"]
+
+match self.profile().arch:
+    case "x86_64":
+        make_build_args.append("ARCH=x86-64")
+    case "ppc":
+        make_build_args.append("ARCH=ppc-32")
+    case "ppc64" | "ppc64le":
+        make_build_args.append("ARCH=ppc-64")
+    case "aarch64":
+        make_build_args.append("ARCH=armv8")
+    case "riscv64":
+        make_build_args.append("ARCH=riscv64")
+    case _:
+        broken = "arch not supported"
```

Comment:
this is a bit of a meme because nothing in the makefile actually does anything with these values. they just pass some not very interesting compiler flags and that's it. it doesn't seem to actually be arch restricted..

the same kind of thing also goes for an lto section in it. really the whole makefile is a mess. maybe you feel like just removing this check since it's not really needed?

if not, you should set archs= above too.

Reply:
I guess we can skip arch specific stuff without patching (SUPPORTED_ARCH)